### PR TITLE
remove peer

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @qiuyesuifeng @disksing

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

File: server/command.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ func (c *conn) handleAskSplit(req *pdpb.Request) (*pdpb.Response, error) {
 	if request.Region == nil {
 		return nil, errors.New("missing region for split")
 	}
-	if request.Leader == nil {
+	if request.LeaderStoreId == nil {
```
Comment (qiuyesuifeng) [inline server/command.go]: Should be `!= 0` ?

Comment (siddontang) [inline server/command.go]: em, the check should be `GetLeaderStoreId != 0`

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**